### PR TITLE
[Feature] Notification - application status changed

### LINKED ISSUE #9642: ❗[Notification] Application Status Changed
## ✨ Purpose

Inform applicants of changes to their application status.

## ❗ Trigger

Final decision or removed recorded by pool operator. 
Reverting included

## 👀 Audience

Only the applicant themselves should receive a notification. 

### Email templates (English and French)

**EN Email Subject** 
 
GC Digital Talent application status updated 

**EN Email body** 

Hey {{user.firstname}}, 

You applied to {{Job Title - links to advertisement}} on GC Digital Talent. We want to let you know that there’s an update to your status. To view the new status, {{visit your profile } {link: track application section of dashboard}}. 

_Email footer_

**FR Email Subject** 
Mise à jour de l’état d’avancement de votre demande sur Talents numériques du GC

**FR Email body** 
Vous avez postulé à [____](url) sur Talents numériques du GC. L’état d’avancement de votre demande a été modifié. Pour voir son nouvel état, veuillez [consulter votre profil](url).

_Email footer_

### In-app string (English and French)

EN In-app Notification copy:  

The status of your application for {{opportunity.title I profile-dashboard}} has been updated. 

EN In-app Notification copy:  
L’état d’avancement de votre demande pour {{opportunity.title}} a été mis à jour.

### Link

Link to the Track your applications page (dashbaord)

## 💾 Content

- What Notification Family does this belong to? (see #9555)

## ✅ Acceptance Criteria

- [ ] Email templates exists in GC Notify
- [ ] Notification classes exists in Laravel
- [ ] Notification is triggered by appropriate event and send to appropriate users (including NOT any users who are ignoring this notification's Notification Family)
- [ ] Notification type(s) added to schema
- [ ] Notification type(s) rendered appropriately on frontend
- [ ] Tests?
- [ ] Remove existing experimental status notification

## 🛑 Blockers

Issues which must be completed before this one.

```[tasklist]
### Blocked By
- [ ] https://github.com/GCTC-NTGC/gc-digital-talent/issues/9640
```

### PR DESCRIPTION
**Pull request by petertgiles**
🤖 Resolves #9642 

## 👋 Introduction

This branch adds a new notification - application status changed - for both in-app and email.

## 🕵️ Details

Add any additional details that could assist with reviewing or testing this PR.

## 🧪 Testing

- set your GCNotify API key (team) and template IDs
- Rebuild and log in
- Setup your applicant user
  - enable in-app and email notifications for the APPLICATION_UPDATE notification family
  - set your email address to your TBS email
  - set your language preference
- submit an application to a pool
- as an admin user, set that application to a "final decision" or "removed" status
- ensure the applicant received an email and an in-app notification

## 📸 Screenshot

image
image

## 🚚 Deployment

- add new environment variables for the two new templates:
  - GCNOTIFY_TEMPLATE_APPLICATION_STATUS_CHANGED_EN
  - GCNOTIFY_TEMPLATE_APPLICATION_STATUS_CHANGED_FR

### CODE REVIEW

**Comment by petertgiles** [comment]
@gobyrne or @marc-donofrio could you please review the email templates in GCNotify?

File: api/app/Notifications/ApplicationStatusChanged.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            ],
+        ];
+    }
+
+    /**
+     * Get the GC Notify representation of the notification.
+     */
+    public function toGcNotifyEmail(User $notifiable): GcNotifyEmailMessage
+    {
+        $locale = $this->locale ?? $notifiable->preferredLocale();
```
Comment (tristan-orourke) [inline api/app/Notifications/ApplicationStatusChanged.php:67]: Where does `$this->locale` come from?

Comment (petertgiles) [reply api/app/Notifications/ApplicationStatusChanged.php:67]: The notification can have its locale explicitly set when it is created.
https://laravel.com/docs/10.x/notifications#localizing-notifications

File: api/app/Observers/PoolCandidateObserver.php [hunk 2]
```diff
@@ -21,6 +25,30 @@ public function updated(PoolCandidate $poolCandidate): void
     {
         $oldStatus = $poolCandidate->getOriginal('pool_candidate_status');
         $newStatus = $poolCandidate->pool_candidate_status;
+        if (config('feature.notifications')) {
+            if (
```
Comment (tristan-orourke) [inline api/app/Observers/PoolCandidateObserver.php:29]: I think this should also check that $newStatus !== $oldStatus

File: api/tests/Feature/ApplicantFilterTest.php [hunk 3]
```diff
@@ -35,7 +36,7 @@ class ApplicantFilterTest extends TestCase
     protected function setUp(): void
     {
         parent::setUp();
-
+        Notify::spy(); // don't send any notifications
```
Comment (tristan-orourke) [inline api/tests/Feature/ApplicantFilterTest.php:39]: So is this something we should add to all Test classes now, to ensure we don't accidentally send notifications? Maybe we should add this to a base class.

File: api/app/Observers/PoolCandidateObserver.php (hunk 2 above)
Comment (petertgiles) [reply api/app/Observers/PoolCandidateObserver.php:29]: Oh yeah!  Good catch.
[only trigger when status changes](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10372/commits/888403d2e94d5140a3837eb240b3f931176cf661)

**Review by tristan-orourke — APPROVED** [review]
Wonderful! Works as expected.

I noticed one little hicough, but I think its out of scope for this one. When I clicked the in-app notification, which links to http://localhost:8000/en/applicant#track-applications-section, and I was already on the page http://localhost:8000/en/applicant, then I wasn't scrolled to the correct section.

**Pull request merged**